doc/cli_guidelines.md

```diff
@@ -0,0 +1,196 @@
+
+$ dinstaller config set <key>=<value> ...
+Sets a config value for the given key.
+
+$ dinstaller config unset <key>
+Removes the current value for the given key.
+
+$ dinstaller config reset [<key>]
+Sets the default value for the given <key>. If no key is given, then the whole config is reset.
+
+$ dinstaller config add <list-key> [<key>=]<value> ...
+Adds a new entry with all the given key-value pairs to a config list. The key is omitted for a list of scalar values (e.g., languages).
+
+$ dinstaller config delete <list-key> [<key>=]<value> ...
+Deletes any entry matching all the given key-value pairs from a config list. The key is omitted for a list of scalar values.
+
+$ dinstaller config check
+Validates the config and prints errors
+
+$ dinstaller config info <key> [<value>]
+Prints info about the given key. If no value is given, then it prints what values are admitted by the given key. If a value is given, then it shows extra info about such a value.
+
+$ dinstaller summary [<section>]
+Prints a summary with the actions to perform in the system. If a section is given (e.g., storage, software, ...), then it only shows the section summary.
+
+$ dinstaller questions
+Prints questions and allows to answer them.
+
+~~~
+
+In those commands `<key>` represents a YAML key from the config structure and `<value>` is the value associated to the given key.
+
+### YAML Config
+
+The config settings are defined by this YAML structure:
+
+~~~
+---
+product: "Tumbleweed"
+
+languages:
+  - "es_ES"
+  - "en_US"
+
+user:
+  name: "test"
+  fullname: "User Test"
+  password: "n0ts3cr3t"
+  autologin: true
+
+root:
+  ssh_key: "1234abcd"
+  password: "n0ts3cr3t"
+
+storage:
+  candidate_devices:
+    - /dev/sda
+  lvm: true
+  encryption_password: 12345
+  volumes:
+    - mountpoint: /
+      fstype: btrfs
+    - mountpoint: /home
+      fstype: ext4
+      minsize: 10GiB
+~~~
+
+Nested keys are referenced in commands by joining them with dots (e.g., `user.name`).
+
+### Examples
+
+Let's see some examples:
+
+~~~
+# Set a product
+$ dinstaller config set product=Tumbleweed
+
+# Set user values
+$ dinstaller config set user.name=linux
+$ dinstaller config set user.fullname=linux
+$ dinstaller config set user.password=linux
+$ dinstaller config set user.name=linux user.fullname=linux user.password=12345
+
+# Unset user
+$ dinstaller config unset user
+
+# Add and delete languages
+$ dinstaller config add languages en_US
+$ dinstaller config delete languages en_US
+
+# Set storage settings
+$ dinstaller config set storage.lvm=false
+$ dinstaller config set storage.encryption_password=12345
+
+# Add and delete candidate devices
+$ dinstaller config add storage.candidate_devices /dev/sda
+$ dinstaller config delete storage.candidate_devices /dev/sdb
+
+# Add and delete storage volumes
+$ dinstaller config add storage.volumes mountpoint=/ minsize=10GiB
+$ dinstaller config delete storage.volumes mountpoint=/home
```

Comment:
to be honest I find this at least a bit confusing. As it can be read as "delete volume where mountpoint is /home" or "delete from volumes key mountpoint with value /home". I know it is described that it should be the first case, but if I see command without reading documentation I will be confused.

Reply:
Umm, `#add` and `#delete` verbs are intended to add/delete an element to/from a list. I see your point but I am not able to propose something more intuitive. Any idea? Anyway, elements in a list are "atomic", you add or delete them, there is no way to modify a single attribute.